kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/rbac/Permission.java

```diff
@@ -25,22 +26,28 @@
 public class Permission {
 
   Resource resource;
+  List<String> actions;
 
   @Nullable
-  Pattern value;
-  List<String> actions;
+  String value;
+  @Nullable
+  transient Pattern compiledValuePattern;
 
   @SuppressWarnings("unused")
   public void setResource(String resource) {
     this.resource = Resource.fromString(resource.toUpperCase());
   }
 
-  public void setValue(String value) {
-    this.value = Pattern.compile(value);
+  @SuppressWarnings("unused")
+  public void setValue(@Nullable String value) {
+    this.value = value;
   }
 
   @SuppressWarnings("unused")
   public void setActions(List<String> actions) {
+    if (actions != null) {
+      actions = actions.stream().map(String::toUpperCase).toList();
```

Comment:
looks redundant due to `.map(String::toUpperCase)` in AccessControlService#isAccessible